tests/std/tests/GH_001103_countl_zero_correctness/test.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <bit>
+#include <cassert>
+
+// Indirectly test countl_zero on old x86/x64 processors by testing private hepler,
+// which is different from the usual branch.
+
+// Since the fallback intrinsic is available on ARM too, don't need to exclude ARM
+
+// Currently need this test only in C++20 mode,
+// May update to older C++ if the helper is used internally too, for example in <bitset>
+
+using namespace std;
+
+int main() {
+#ifdef _CHAR_UNSIGNED

```

Comment:
I believe these are required regardless of the signed-ness of `char`. Also note that our default is that char is signed, so I don't think this block is tested in GitHub right now.

Reply:
count"_" only work for unsigned types, signed will fail test

Reply:
also: all the <bit> functions are sfiane'd out for signed char (and char, I think, probably.... boy char is an odd type.)

Reply:
maybe remove this block altogether?

Reply:
We have an extremely small number of `/J` tests (this is the option that alters the signedness of char) and I don't think it's worth testing further.

Reply:
Looking at the Standard constraint on `countl_zero`, "`T` is an unsigned integer type", that specifically restricts it to be `unsigned char`, `unsigned short`, `unsigned int`, `unsigned long`, and `unsigned long long`. (We have no extended integer types.) So this `char` coverage should be completely removed.